log: add page (#6316)
tldr-pages/tldr: pages/osx/log.md

# log

> > View system wide log messages

Reviewer: ```suggestion
> View system wide log messages.
> More information: <https://ss64.com/osx/login.html>.
```

> - Stream live system logs

Reviewer: ```suggestion
- Stream live system logs:
```

# log

> View, export, and configure logging systems.
> More information: <https://www.dssw.co.uk/reference/log.html>.

- Stream live system logs:

`log stream`

> - Stream logs sent to syslog for PID 235:

Reviewer 2: Perhaps this will read better?
```suggestion
- Stream logs sent to syslog from the process with the PID 235:
```

> - Stream logs sent to syslog from the process with a specific PID:

Reviewer 3: For technical terms in descriptions, we use backticks:
```suggestion
- Stream logs sent to `syslog` from the process with a specific PID:
```

> `log stream --process {{235}}`

Reviewer 4: ```suggestion
`log stream --process {{PID}}`
```

Reviewer: the placeholders should be in `snake_case`.

Reviewer 4: PID is an acronym, therefore it should be in caps (Process ID)

Reviewer: You can use `{{process_id}}` to be more friendly for new users.

Reviewer 2: It is an acronym, but in this case it should be lowercase regardless

Reviewer 4: I like process_id

my suggestion has been edited

> - Show logs sent to syslog for process named 'dasd':

Reviewer 2: Perhaps this will read better?
```suggestion
- Show logs sent to syslog from a process named 'dasd':
```

> `log show --predicate "process == '{{dasd}}'"`

Reviewer 4: ```suggestion
`log show --predicate "process == '{{process_name}}'"`
```

> - Dump all logs for the past hour:

Reviewer 2: Perhaps this will read better?
```suggestion
- Print all logs from the past hour:
```

Author: I agree with everything you said... except this one. :) Since this one saves the logs to disk, "Print" isn't the best verb. How does this work: "Export all logs to disk for the past hour"

Reviewer 2: Yeah, that sounds good @jchilders!

> `sudo log collect --last {{1h}}`

Reviewer 3: Ref: https://github.com/tldr-pages/tldr/pull/6316/files#r688601213
```suggestion
`sudo log collect --last {{1h}} --path {{path/to/file.logarchive}}`
```

> `sudo log collect --last {{1h}} --path {{path/to/file.logarchive}}`

Reviewer: @marchersimon why `.logarchive` instead of `.log`?
```suggestion
`sudo log collect --last {{1h}} --path {{path/to/file.log}}`
```

Reviewer 3: From the man page:
> -output path
>
> Save the archive to the specified path or file. If the path is a directory, a file named system_logs.logarchive will be created in the specified directory. If the path contains the extension .logarchive, a new logarchive will be created with that name at the specified path.

But I just noticed that I confused `output` with `path`. It should be
```suggestion
`sudo log collect --last {{1h}} --output {{path/to/file.logarchive}}`
```

`log stream`

- Stream logs sent to syslog from the process with a specific PID:

`log stream --process {{235}}`

- Show logs sent to syslog from a process with a specific name:

`log show --predicate "process == '{{dasd}}'"`

> - Export all logs to disk for the past hour:

Reviewer 3: From the docs:
> If an output path is not specified, system_logs.logarchive will be created in the current directory.

But I think we should add the `--output` flag to avoid confusion
